### PR TITLE
fix(output): correct unimportant vuln type

### PR DESCRIPTION
**Pull request by hogo6002**
resolves https://github.com/google/osv-scanner/issues/1419

- Print unimportant vulnerabilities as unimportant, not uncalled.
- Add an `IsGroupUnimportant()` function to identify if a vulnerability is unimportant.
- Refactor the output result structure from `CalledVulns` and `UncalledVulns` to `RegularVulns` and `HiddenVulns`.
- Add `VulnAnalysisType` to indicate if a vulnerability is regular, unimportant, or uncalled. This can also be expanded in the future when we have more reachability analysis. 

Project scanning (Changed from "uncalled vulnerabilities" to "unimportant vulnerabilities" for Debian package scanning):
Screenshot 2024-12-06 at 4 17 46 PM
Contianer scanning (Added a "filtered vulnerabilities" table with a filter reason):
Screenshot 2024-12-06 at 4 19 06 PM

### CODE REVIEW

File: internal/output/output_result.go [hunk 1]
```diff
@@ -94,22 +96,30 @@ type SeverityCount struct {
...
+	All     int
+	OS      int
+	Project int
+	Hidden  int
 }
 
-type CallAnalysisCount struct {
-	Called   int
-	Uncalled int
+type AnalysisCount struct {
```
Comment (another-rex) [inline internal/output/output_result.go:105]: Can you add a comment here and above on VulnTypeCount to explain why there is two different counts?

File: internal/output/table.go [hunk 2]
```diff
@@ -134,6 +146,30 @@ func printContainerScanningResult(result Result, outputWriter io.Writer, termina
 			outputTable.Render()
 		}
 	}
+
+	if result.VulnTypeCount.Hidden != 0 {
+		// Add a newline
+		fmt.Fprintln(outputWriter)
+		fmt.Fprintln(outputWriter, "Filtered Vulnerabilities:")
+		outputTable := newTable(outputWriter, terminalWidth)
+		outputTable.AppendHeader(table.Row{"Package", "ecosystem", "Installed Version", "Filtered vuln count", "Filter Reason"})
```
Comment (another-rex) [inline internal/output/table.go:155]:
```suggestion
		outputTable.AppendHeader(table.Row{"Package", "Ecosystem", "Installed Version", "Filtered Vuln Count", "Filter Reasons"})
```

**Review by another-rex — COMMENTED** [review]
Good idea with labelling each count type!

File: internal/output/output_result.go (hunk 1 above)
Comment (hogo6002) [reply internal/output/output_result.go:109]: renamed `VulnTypeCount` to `VulnTypeSummary` and added comments. The `VulnTypeSummary` here is mostly used to implement the HTML vuln type filter.

**Review by another-rex — APPROVED** [review]

**Pull request merged**